docs/examples/acme-issuer.yaml

```diff
@@ -46,3 +46,21 @@ spec:
           # This field is optional for overriding the Route53 hosted zone ID
           # It is required to use it if the cert-manager cannot disambiguate between two different hosted zones for the same zone name
           hostedZoneID: DIKER8JPL21PSA
+      - name: azuredns
+        azuredns:
+          # Service principal clientId (also called appId)
+          clientID: 8ff041f4-a14f-4753-80c2-101b35db5879
+          # A secretKeyRef to a service principal ClientSecret (password)
+          # ref: https://docs.microsoft.com/en-us/azure/container-service/kubernetes/container-service-kubernetes-service-principal
+          clientSecretSecretRef:
+            name: azuredns-config
+            key: client-secret
+          # Azure subscription Id
+          subscriptionID: 0933cdcc-0cd0-4fb3-9f26-dac4fdc2154b
+          # Azure AD tenant Id
+          tenantID: 9581f7ad-8f4f-4f07-92df-12c821981ce8
```

Comment:
There are a *lot* of different parameters for connecting to Azure!

Given #197, I think we are slowly moving toward storing all of this information in secret resources. Perhaps we should also make *another* breaking API change and condense them all into a single secret, removing the `key` field and instead forcing users to use particular predefined (and documented) keys. This might be a painful change, but better to do it sooner rather than later.

Reply:
Yes, I know, a lot...

> Given #197, I think we are slowly moving toward storing all of this information in secret resources. 

Do you think about putting clientId into secret or also tenantId, subscriptionId, etc? #197 is only about accessKey/secretKey.
Let me know what do you think and if you would like to change it in this PR or another one in the future.

Reply:
@munnerz If you find time to look at it - please let me know if I should change anything related to comments above. thx